tests/auth_tests/test_auth_backends.py

```diff
@@ -803,6 +814,126 @@ def test_clean_credentials_sensitive_variables(self):
             status_code=500,
         )
 
+    @override_settings(
+        ROOT_URLCONF="django.contrib.auth.urls",
+        AUTHENTICATION_BACKENDS=["auth_tests.test_auth_backends.TypeErrorBackend"],
+    )
+    def test_login_process_sensitive_variables(self):
+        try:
+            self.client.post(
+                reverse("login"),
+                dict(username="testusername", password=self.sensitive_password),
+            )
+        except TypeError:
+            exc_info = sys.exc_info()
+
+        rf = RequestFactory()
+        response = technical_500_response(rf.get("/"), *exc_info)
+
+        self.assertNotContains(response, self.sensitive_password, status_code=500)
+        self.assertContains(response, "TypeErrorBackend", status_code=500)
+
+        # Check AuthenticationForm.clean function
+        self.assertContains(
+            response,
+            '<tr><td>password</td><td class="code">'
+            "<pre>&#39;********************&#39;</pre></td></tr>",
+            html=True,
+            status_code=500,
+        )
+
+        # Check Client.post function
+        self.assertContains(
+            response,
+            '<tr><td>data</td><td class="code">'
+            "<pre>&#x27;********************&#x27;</pre></td></tr>",
+            html=True,
+            status_code=500,
+        )
+
+        # Check RequestFactory.post function
+        self.assertContains(
+            response,
+            '<tr><td>post_data</td><td class="code">'
+            "<pre>&#x27;********************&#x27;</pre></td></tr>",
+            html=True,
+            status_code=500,
+        )
+
+    def test_setpasswordform_validate_passwords_sensitive_variables(self):
+        try:
+            password_form = SetPasswordForm(AnonymousUser())
+            password_form.cleaned_data = {
+                "password1": "test_password1",
+                "password2": "test_password2",
+            }
+            password_form.validate_passwords()
+        except ValueError:
+            exc_info = sys.exc_info()
+
+        rf = RequestFactory()
+        response = technical_500_response(rf.get("/"), *exc_info)
+
```

Comment:
```suggestion
        response = technical_500_response(rf.get("/"), *exc_info)
        self.assertNotContains(response, "test_password1", status_code=500)
        self.assertNotContains(response, "test_password2", status_code=500)
```
When I add the following, this test fails - I would expect this to pass